### PR TITLE
square.go

### PR DESCRIPTION
**Pull request by egor-geller**

### CODE REVIEW

File: square.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	y1 = e.start.y + int(e.a)
+	return x1, y1
+}
+
+func (p Square) Perimeter() (p1 int)  {
+	return int(p.a)*4
+}
+
+func (s Square) Area() (s1 int) {
+	return int(s.a)*2
```
Comment (burov) [inline square.go]: I think you don't have to explicitly cast to int s.a, because it's already an int

Comment (burov) [reply square.go]: If I'm not mistaking, area of the Square, it's a s.a * s.a, not a s.a * 2

File: square.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+type Point struct {
+	x, y int
+}
+
+type Square struct {
+	start Point
+	a     uint
+}
+
+func (e Square) End() (x1, y1 int){
```
Comment (burov) [inline square.go]: function receivers should be consistent, use only ```func (s Square) End() (x1, y1 int){``` for all

File: square.go (hunk 1 above)
Comment (egor-geller) [reply square.go]: Yes forgot about the area, s.a is a uint, i can cast it as an int or change the return value (for example s1) to uint, no?

Comment (burov) [reply square.go]: Oh, I see you have to cast s.a to int, both options acceptable it's depends on specific case.  :)

**Review by burov — APPROVED** [review]

File: square.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+type Point struct {
+	x, y int
+}
+
+type Square struct {
+	start Point
+	a     uint
+}
+
+func (s Square) End() (x1, y1 int){
```
Comment (imatsiash-ib) [inline square.go]: This method should return `Point` structure.

**Review by imatsiash-ib — APPROVED** [review]

**Pull request closed without merging**